java/org/purefn/river/Message.java

```diff
@@ -13,14 +13,26 @@ private Object fetchKey(Object key) {
 
 	Keyword k = (Keyword) key;
 	switch (k.getName()) {
```

Comment:
This repetition makes javascript look good. Isn't there a way to make this function more expressive, or do we expect there to be cases that do not fit this pattern?

Reply:
Sorry, this is Java (not Clojure), so the answer is probably no.

Reply:
`switch` is the most performant, performance being the goal here, not elegance of code.